pkg/kv/kvserver/rangefeed/scheduler.go

```diff
@@ -300,9 +300,9 @@ func (s *Scheduler) register(id int64, f Callback, priority bool) error {
 // Any attempts to enqueue events for processor after this call will return an
 // error.
 func (s *Scheduler) unregister(id int64) {
-	_, priority := s.priorityIDs.Load(id)
+	priority := s.priorityIDs.Contains(id)
```

Comment:
This isn't quite threadsafe. It call `Contains` but two rows down call `Remove`. The entry could have been removed in-between. It would be better to call Remove first and then call `s.shards.unregister` if Remove returns `true`.

This didn't change with your commit and may not matte in a meaningful way, but it would be nice to fix it.

Reply:
Agreed that this looks sketchy on first glance. I dug in more and it's less of an issue than it initially seems. The set is just used to locate the shard that the corresponding ID is in. An ID may or may not be in the priorityIDs set, depending on whether the ID has "priority".

The reason this isn't actually broken is because:
1. an ID is only used once, so we can't have interleavings of `register` and `unregister` with the same IDs.
2. `unregister` is only ever called once for a given ID.
3. `enqueue` gracefully ignores the missing ID case.